v2/internal/genericarmclient/cloud_error.go

```diff
@@ -27,4 +49,54 @@ func (e CloudError) Error() string {
 	return e.error.Error()
 }
 
-func (e CloudError) Unwrap() error { return e.error }
+// ErrorCode returns the error code from the message.
+// If InnerError is present, we have an ARM compliant error (our preferred kind), so we use that code if present.
+// If InnerError is not present, we return the outer error code if present.
+// If neither is present, we return UnknownErrorCode
+func (e CloudError) ErrorCode() string {
```

Comment:
Did you consider overriding the `UnmarshalJSON` handler and deserializing both flavors into a single structure? The advantage would be:
- No duplicate fields
- No need to write accessor methods (which are fine but awkward when the fields they are exposing are public and anybody can decide to read from them, which might work most of the time but be wrong)

Reply:
I implemented `UnmarshalJSON()` but kept the accessor methods as they do useful things (like handling **nil** values) that make consumption of the values simpler.